assets/js/app/Instance.js

```diff
@@ -264,19 +262,23 @@ LocusZoom.Instance.prototype.mapTo = function(chr, start, end){
         this.remap_promises.push(this.panels[id].reMap());
     }
 
-    // When all finished update download SVG link
-    Q.all(this.remap_promises).then(function(){
-        this.controls.setBase64SVG();
-    }.bind(this), function(error){
-        console.log(error);
-        this.curtain.drop(error);
-    }.bind(this));
+    Q.all(this.remap_promises)
+        .then(this.mapToComplete)
+        .catch(function(error){
+            console.log(error);
+            this.curtain.drop(error);
+        }.bind(this));
```

Comment:
I've found that adding a `.done()`  at the end of a Q promise can be helpful for debugging. Even with at `.catch()`, there may be an error in that function and the `.done()` helps to give more useful error messages. Of course It should only be used at the terminal end of a promise chain.